Problem compiling lapacke and boost together

The following test example:
```c++
#include "lapacke.h"
#include "boost/property_tree/ptree.hpp"

int main() {
    return 0;
}
```
does not compile, with errors like this:
```
In file included from /usr/include/c++/7/complex.h:36:0,                                                                                                                                       
                 from /home/xap/local/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.3.0/openblas-0.3.3-xlt4harsqysny4q4qw53g34s6jfz5wtx/include/lapacke.h:73,                             
                 from ./main.cc:1:                                                                                                                                                             
/home/xap/local/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.3.0/boost-1.68.0-j2mqoxflxrkb5c27hqgzp3yer5aepbtp/include/boost/type_traits/integral_constant.hpp:34:20: error: expected identifier before ‘(’ token                                                                                                                                                                       
    template <class I, I val> struct integral_c;                                                                                                                                               
                    ^                                                                                                                                                                          
/home/xap/local/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.3.0/boost-1.68.0-j2mqoxflxrkb5c27hqgzp3yer5aepbtp/include/boost/type_traits/integral_constant.hpp:34:20: error: expected ‘)’ before ‘__extension__’                                                                                                                                                                        
    template <class I, I val> struct integral_c;                                                                                                                                               
                    ^                                                                                                                                                                          
/home/xap/local/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.3.0/boost-1.68.0-j2mqoxflxrkb5c27hqgzp3yer5aepbtp/include/boost/type_traits/integral_constant.hpp:34:20: error: expected ‘>’ before ‘__extension__’                                                                                                                                                                        
/home/xap/local/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.3.0/boost-1.68.0-j2mqoxflxrkb5c27hqgzp3yer5aepbtp/include/boost/type_traits/integral_constant.hpp:34:20: error: expected unqualified-id before ‘)’ token                                                                                                                                                                   
    template <class I, I val> struct integral_c;
```
This is also a problem with many other boost headers as well.

The only known workaround is to change the order so that `lapacke.h` goes *after* boost header. It sounds like a bug somewhere, but I have trouble understanding why that is the case. It always go through `complex.h` header, but unclear how and why it ends with boost.

Also, once I reordered with boost in my project, it got triggered on Kokkos library too:
```
In file included from /usr/include/c++/7/complex.h:36:0,                                                                                                                                       
                 from /home/xap/local/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.3.0/openblas-0.3.3-xlt4harsqysny4q4qw53g34s6jfz5wtx/include/lapacke.h:73,                             
                 from ../../include/mfmg/dealii/lanczos_lanczos.templates.hpp:22,                                                                                                              
                 from ../../include/mfmg/dealii/lanczos_deflatedlanczos.templates.hpp:23,                                                                                                      
                 from ../../tests/test_lanczos.cc:14:                                                                                                                                          
/home/xap/local/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.3.0/trilinos-master-wahhgsi2mnozyjqlhbgn6wyynupjs2q3/include/impl/Kokkos_Utilities.hpp: At global scope:                   
/home/xap/local/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.3.0/trilinos-master-wahhgsi2mnozyjqlhbgn6wyynupjs2q3/include/impl/Kokkos_Utilities.hpp:160:20: error: expected ‘)’ before ‘_
_extension__’                                                                                                                                                                                  
 template <unsigned I, typename IntegerSequence>          
```
Both boost and Kokkos seem to be triggered when using `I` in a template. So I'm no longer certain that's just boost. But it is almost certain have something to do with complex.

--- a/lapack-netlib/LAPACKE/include/lapacke.h
+++ b/lapack-netlib/LAPACKE/include/lapacke.h
@@ -86,7 +86,11 @@ lapack_complex_float lapack_make_complex_float( float re, float im );
 
 /* Complex type (double precision) */
 #ifndef lapack_complex_double
+#ifndef __cplusplus
 #include <complex.h>
+#else
+#include <complex>
+#endif
 #define lapack_complex_double   double _Complex
 #endif
 

